plugins/modules/foreman_compute_profile.py

```diff
@@ -121,97 +121,49 @@
-        find_compute_profile,
-        find_compute_attribute,
-        ForemanEntityAnsibleModule,
-        naildown_entity,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
-
-    from nailgun.entities import (
-        AbstractComputeResource,
-        ComputeProfile,
-        ComputeAttribute,
-    )
-except ImportError:
-    pass
-
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
 
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-}
 
-compute_attribute_name_map = {
-    'compute_resource': 'compute_resource',
-    'vm_attrs': 'vm_attrs',
+compute_attribute_entity_spec = {
+    'compute_resource': {'type': 'entity', 'flat_name': 'compute_resource_id'},
+    'vm_attrs': {'type': 'dict'},
 }
 
 
-def compute_attribute(module, compute_profile, attributes):
-    # If we don't work on a copy, we hit a TypeError: Value of unknown type: <class 'nailgun.entities.ComputeProfile'> exception
-    # from module.exit_json(changed=changed) method
-    compute_attribute_dict = attributes.copy()
-    for key in ['compute_resource', 'vm_attrs']:
-        if key not in compute_attribute_dict:
-            module.fail_json(msg='compute_attribute must have %s.' % key)
-
-    try:
-        compute_attribute = find_compute_attribute(module, compute_profile_name=compute_profile.name,
-                                                   compute_resource_name=compute_attribute_dict['compute_resource'], failsafe=True)
-        compute_attribute_dict['compute_resource'] = find_compute_resource(module, name=compute_attribute_dict['compute_resource'])
-    except Exception as e:
-        module.fail_json(msg='Failed to find entity: %s ' % e)
-
-    if 'compute_profile' not in compute_attribute_dict:
-        compute_attribute_dict['compute_profile'] = compute_profile
-
-    changed = naildown_entity_state(ComputeAttribute, compute_attribute_dict, compute_attribute, 'present', module)
-
-    return changed
-
-
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
+        entity_spec=dict(
             name=dict(required=True),
+            compute_attributes=dict(type='nested_list', entity_spec=compute_attribute_entity_spec),
+        ),
+        argument_spec=dict(
             updated_name=dict(),
-            compute_attributes=dict(type='list'),
         ),
     )
 
-    (compute_profile_dict, state) = module.parse_params()
-    name = compute_profile_dict.get('name')
-    updated_name = compute_profile_dict.get('updated_name')
-    compute_attributes = compute_profile_dict.pop('compute_attributes', list())
-
-    if len(compute_attributes) > 0 and state == 'absent':
-        module.fail_json(msg='compute_attributes not allowed when state=absent')
+    entity_dict = module.clean_params()
+    updated_name = entity_dict.get('updated_name')
+    compute_attributes = entity_dict.pop('compute_attributes', None)
```

Comment:
Any reason to have the fallback `None` instead of `[]` (the spec says it's a list…)